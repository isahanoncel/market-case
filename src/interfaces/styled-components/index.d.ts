// #region Global Imports
import 'styled-components';
// #endregion Global Imports

export type Icons = 'Bag' | 'ArrowLeft' | 'Check';

export type Colors =
  | 'primary'
  | 'gray'
  | 'secondary'
  | 'white'
  | 'background'
  | 'transparent'
  | 'innerBorder'
  | 'black100'
  | 'black600';

type Fonts = 'OpenSans';

declare module 'styled-components' {
  export interface DefaultTheme {
    colors: Record<Colors, string>;
    fonts: Record<Fonts, string>;
  }
}
