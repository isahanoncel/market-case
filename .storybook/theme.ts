import { create } from '@storybook/theming';

export default create({
  base: 'light',
  brandTitle: 'Getir',
  brandUrl: 'https://getir.com',
  brandTarget: '_self',
});
