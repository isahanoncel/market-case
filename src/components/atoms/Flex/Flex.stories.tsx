import React from 'react';
import { ComponentMeta, ComponentStory } from '@storybook/react';
import Flex from '.';

export default {
  title: 'Atoms/Flex',
  component: Flex,
} as ComponentMeta<typeof Flex>;

export const Default: ComponentStory<typeof Flex> = (args) => (
  <Flex {...args}>Flex</Flex>
);
