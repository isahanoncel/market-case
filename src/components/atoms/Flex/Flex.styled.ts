import styled from 'styled-components';
import { device } from '../../../constants/styles/GlobalStyle';

export interface IFlexContainer {
  justifyContent?: 'center' | 'flex-end' | 'flex-start' | 'space-between';
  alignItems?: 'center' | 'flex-end' | 'flex-start';
  flexDirection?: 'column' | 'row';
  flex?: number;
  margin?: string;
}

export const FlexContainer = styled.div<IFlexContainer>`
  display: flex;
  height: 100%;
  ${device.mobile} {
    height: auto;
  }
  ${(props) =>
    props.justifyContent && `justify-content:${props.justifyContent}`};
  ${(props) => props.alignItems && `align-items:${props.alignItems}`};
  ${(props) => props.flexDirection && `flex-direction:${props.flexDirection}`};
  ${(props) => props.flex && `flex: ${props.flex}`};
  ${(props) => props.margin && `margin: ${props.margin}`};
`;
