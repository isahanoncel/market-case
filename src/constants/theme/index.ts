import { DefaultTheme } from 'styled-components';

export const COLORS = {
  white: '#fff',
  primary: '#1EA4CE',
  whiteBlue: '#F2F0FD',
  secondary: '#147594',
  black100: '#E0E0E0',
  black500: '#6F6F6F',
  black600: '#525252',
  gray: '#697488',
  background: '#E5E5E5',
  transparent: 'transparent',
  innerBorder: '#DFDEE2',
  thinGray: '#A8A8A8',
};

export const FONTS = {
  OpenSans: "'Open Sans', sans-serif",
};

const heroTheme: DefaultTheme = {
  colors: COLORS,
  fonts: FONTS,
};

export default heroTheme;
