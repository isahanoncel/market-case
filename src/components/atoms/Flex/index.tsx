import React, { FC } from 'react';
import * as S from './Flex.styled';

interface IFlex extends S.IFlexContainer {
  children: React.ReactNode | string;
  onClick?:() => void;
  externalCss?:React.CSSProperties
}
const Flex: FC<IFlex> = ({
  children,
  justifyContent,
  alignItems,
  flexDirection,
  margin,
  flex,
  onClick,
  externalCss
}) => {
  return (
    <S.FlexContainer
      style={externalCss}
      onClick={onClick}
      justifyContent={justifyContent}
      alignItems={alignItems}
      flexDirection={flexDirection}
      flex={flex}
      margin={margin}
    >
      {children}
    </S.FlexContainer>
  );
};

export default Flex;
