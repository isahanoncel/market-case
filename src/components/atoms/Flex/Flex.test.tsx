import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import Flex from '.';
import theme from '../../../constants/theme';

test('Flex snapshot testing', () => {
  const tree = renderer
    .create(
      <ThemeProvider theme={theme}>
        <Flex>Flex</Flex>
      </ThemeProvider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
