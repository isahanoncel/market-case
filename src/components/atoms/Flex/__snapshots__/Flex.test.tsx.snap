// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Flex snapshot testing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

@media (min-width:0px) and (max-width:768px) {
  .c0 {
    height: auto;
  }
}

<div
  className="c0"
>
  Flex
</div>
`;
